rwengine/src/render/GameShaders.cpp

```diff
@@ -236,12 +236,14 @@ float alphaThreshold = (1.0/255.0);
 void main()
 {
 	// Only the visibility parameter invokes the screen door.
-	vec4 diffuse  = texture(tex, TexCoords);
+	vec4 diffuse = Colour;
+	diffuse.rgb += ambient.rgb*ambientfac;
+	diffuse *= colour;
+	diffuse *= texture(tex, TexCoords);
 	if(visibility <= filterMatrix[int(gl_FragCoord.x)%4][int(gl_FragCoord.y)%4]) discard;
 	if(diffuse.a <= alphaThreshold) discard;
-	vec4 colour_dyn = diffuse * colour;
 	float fog = 1.0 - clamp( (fogEnd-WorldSpace.w)/(fogEnd-fogStart), 0.0, 1.0 );
-	fragOut = vec4(mix(colour_dyn.rgb * dynamic.rgb + ambient.rgb * ambient.a, fogColor.rgb, fog), colour_dyn.a);
+	fragOut = vec4(mix(diffuse.rgb, fogColor.rgb, fog), diffuse.a);
```

Comment:
I felt the colors were too strong (at least compared to what I remember).

So I looked into it:
```
out = (ambientLight*ambientProperty + diffuseLight*diffuseProperty + prelighting)*materialColor
```
(given this from your reference - which is also by you, but it's fine)

Assumption:
```
ambientLight = ambient.rgb
ambientProperty = ambientfac
diffuseLight = Colour
diffuseProperty = colour
prelighting = 0.0 <- not supported yet?!
materialColor = texture(tex, TexCoords)
```

Wouldn't the proposed shader do:
```
out_intended = (ambientLight*ambientProperty + diffuseLight*diffuseProperty + prelighting)*materialColor
out_shader   = (ambientLight*ambientProperty * diffuseLight*diffuseProperty + prelighting)*materialColor
                                             ^--- Bug?
```